### PR TITLE
feat: whitelist addresses param for setting fee tokens

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

Having the capability to designate a new token as a whitelisted fee token on the same day it becomes active on Osmosis is strategically important. Currently, we must wait a week for a governance process to include this token in the whitelist. To address this, we propose introducing a new parameter. This parameter would function similarly to the existing taker fee whitelist parameter, enabling a selected list of specific addresses the authority to modify the list of whitelisted fee tokens.

## Testing and Verifying

Added respective gotests to newly added methods.

## Summary by CodeRabbit

- **New Features**
	- Introduced a feature to whitelist addresses for setting fee tokens, enhancing security and control over transaction fee parameters.
	- Added a new `Params` message in the `txfees` module to hold parameters including whitelisted fee token setters.
	- Implemented a service with a method for setting fee tokens, accessible only to whitelisted addresses.
- **Bug Fixes**
	- Addressed issues in the CometBFT project including batch verification, commit signature verification, ABCI response saving, and rollback command functionality.
- **Documentation**
	- Updated CHANGELOG.md with details on new features and bug fixes in the Osmosis and CometBFT projects.

### CODE REVIEW

File: CHANGELOG.md
Comment (czarcas7ic) [inline CHANGELOG.md:97]: Drive by change

**Review by PaddyMc — APPROVED** [review]
:zap:

File: CHANGELOG.md
Comment (PaddyMc) [reply CHANGELOG.md:97]: Added the update in a separate PR.

https://github.com/osmosis-labs/osmosis/pull/7865

File: x/txfees/keeper/feetokens.go
Comment (p0mvn) [inline x/txfees/keeper/feetokens.go:201]: nit: could we use a helper here?

https://github.com/osmosis-labs/osmosis/blob/907fce5c98bb2aed9b289542aac7ee75918b22b9/osmoutils/slice_helper.go#L97-L105

File: x/txfees/keeper/msg_server.go
Comment (p0mvn) [inline x/txfees/keeper/msg_server.go:22]: Would it make sense to have `AddFeeTokens` and `RemoveFeeTokens`

I think that the current UX of having to repeat all fee tokens is tedious and risky because it is easy to miss a previously existing token

**Review by p0mvn — APPROVED** [review]
LGTM

I think that having `AddFeeTokens` and `RemoveFeeTokens` would be worthwhile additions but not blocking on this

File: x/txfees/keeper/feetokens.go
Comment (czarcas7ic) [reply x/txfees/keeper/feetokens.go]: Use helper here https://github.com/osmosis-labs/osmosis/pull/7855/commits/9cad1cd5d90982bb5b5ab289eb9cfc08397b7df6

File: x/txfees/keeper/msg_server.go
Comment (czarcas7ic) [reply x/txfees/keeper/msg_server.go:22]: @p0mvn  This acts as both a add and remove fee tokens, this should not require passing in all previously existing tokens. I will wait for you to ACK that I am not misunderstanding before I merge

Comment (czarcas7ic) [reply x/txfees/keeper/msg_server.go:22]: @p0mvn this method iterates over the setFeeToken call here https://github.com/osmosis-labs/osmosis/blob/00b080a665874350ad82530124b28ecb3ce88921/x/txfees/keeper/feetokens.go#L136-L158

If the token's poolId is set to 0, it is deleted. If it is new, it is added to the prefix store. Therefore this method acts as both an AddFeeToken and RemoveFeeToken, and there is no requirement to provide the entire fee token set each time. Please ACK that you agree and then I will merge.

Comment (p0mvn) [reply x/txfees/keeper/msg_server.go:22]: ACK.

I meant something else - to make sure that we don't have to specify all the tokens given the "set" abstraction. That is, that "Set" does not accidentally remove tokens not listed as part of the input.

Upon reviewing the code, this is not a problem.

**Pull request merged**
